src/daft-plan/src/logical_ops/join.rs

```diff
@@ -167,6 +169,35 @@ impl Join {
         }
     }
 
+    fn deduplicate_exprs(exprs: Vec<Arc<Expr>>) -> Vec<Arc<Expr>> {
+        let mut counts: HashMap<Arc<Expr>, usize> = HashMap::new();
+
+        exprs
+            .into_iter()
+            .map(|expr| {
+                let count = counts.entry(expr.clone()).or_insert(0);
+                *count += 1;
+
+                if *count == 1 {
+                    expr // First occurrence, return the original expression
+                } else {
+                    let unique_id = Uuid::new_v4();
```

Comment:
We don't necessarily need to use UUID here. We could just rename all of the expressions in order.

Reply:
Indeed, that was my first iteration. However, consider a case where a dataframe has columns named as 'a', 'a1'...(unlikely, but possible), or any other order a user picked for their data. There is no fixed way to ensure that the renamed column names do not clash with another column mistakenly. Does that make sense ? What could be a possible renaming strategy ?

Reply:
What if you renamed all of the expressions?

Reply:
So renaming all expressions (without UUID) causes multiple tests to fail(code snippet below). 

fn rename_join_keys(exprs: Vec<Arc<Expr>>) -> Vec<Arc<Expr>> {
        exprs
            .into_iter()
            .enumerate()
            .map(|(index, expr)| { 
                expr.alias(format!("alias_{}", index)) // Use index to rename join keys
            })
            .collect()
    }
    

Also digging into the join semantics before any of the changes, I see that aliasing all join keys changes the data frame as follows (see screenshot): an extra right.A column is added as there is no matching key name on the left data frame. If the user aliases the join key explicitly, they know what they want. However, changing the join keys under the hood gives them the extra column which they may or may not want. What should be the expected behavior in this case ?

<img width="758" alt="daft_join_keys_alias" src="https://github.com/user-attachments/assets/1fe8aad2-4345-464b-9535-f9c60b36fd23">

Reply:
Ah you're right. We use column expressions with common names to infer the common join keys. However I realize your current implementation may have the same issue. For example, if you do 
```python
df1.join(df2, left_on=["b", "a"], right_on=["a", "a"])
```
Will it return a dataframe with two columns "a" and "b", or would it also have a "right.a"?

Reply:
This is starting to get a little complicated but what you may need to do is iterate through both left_on and right_on, first get the common join keys, put those into your hashmaps, and then go through the expressions again and rename them only if they already exist in the hashmap and they are not common join keys

Reply:
> Ah you're right. We use column expressions with common names to infer the common join keys. However I realize your current implementation may have the same issue. For example, if you do
> 
> ```python
> df1.join(df2, left_on=["b", "a"], right_on=["a", "a"])
> ```
> 
> Will it return a dataframe with two columns "a" and "b", or would it also have a "right.a"?

My UUID solution doesnt have the problem because, I provide aliases only from the second occurrence.

Reply:
The ordering is the issue here, since it is the second join key that should be merged

Reply:
> The ordering is the issue here, since it is the second join key that should be merged

I see, yes in the case of reordered left columns this can still occur

Reply:
The latest changes should resolve the ordering issue.